spec/services/converters/order_detail_to_journal_rows_spec.rb

```diff
@@ -0,0 +1,55 @@
+require "rails_helper"
+
+RSpec.describe Converters::OrderDetailToJournalRows, type: :service do
+
+  let(:converter) { described_class.new(journal, order_detail) }
+  let(:journal) { build_stubbed(:journal) }
+  let(:order_detail) do
+    build_stubbed(:order_detail).tap do |order_detail|
+      allow(order_detail).to receive_message_chain(:product, :account) { "product_account" }
+      allow(order_detail).to receive(:total) { 100 }
+      allow(order_detail).to receive(:long_description) { "long_description" }
+    end
+  end
+
+  it "initializes journal" do
+    expect(converter.journal).to eq(journal)
+  end
+
+  it "initializes order_detail" do
+    expect(converter.order_detail).to eq(order_detail)
+  end
+
+  describe "#to_a" do
+    let(:returned) { converter.to_a }
```

Comment:
Not sure if it's worth it, but you could delegate `each` to `to_a`, and then `include Enumerable` and you should be able to call `size` and `first` on the converter itself.

I'm also not sure I'm in love with `to_a` as the primary API. I wouldn't mind leaving it as an alias, but I think something like `#convert` would be clearer.